sdk/scala-sdk-protobuf-testkit/src/main/scala/kalix/scalasdk/testkit/EventingTestKit.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.scalasdk.testkit
+
+import akka.annotation.{ ApiMayChange, InternalApi }
+import com.google.protobuf.ByteString
+import kalix.javasdk.testkit.{ EventingTestKit => JEventingTestKit }
+import kalix.scalasdk.Metadata
+import kalix.scalasdk.testkit.impl.TopicImpl
+
+/**
+ * Testkit utility to mock broker's topic. Useful when doing integration tests for services that do eventing (in or out)
+ * to a broker's topic.
+ */
+@ApiMayChange
+trait Topic {
+
+  /**
+   * Waits and returns the next unread message on this topic. Note the message might have been received before this
+   * method was called. If no message is received, a timeout exception is thrown.
+   *
+   * @return
+   *   message including ByteString payload and metadata
+   */
+  def expectNext(): Message[ByteString]
+
+  /**
+   * Waits for a default amount of time before returning all unread messages in the topic. If no message is received, a
+   * timeout exception is thrown.
+   *
+   * @return
+   *   collection of messages, each message including ByteString payload and metadata
+   */
+  def expectAll(): Seq[Message[ByteString]]
+}
```

Comment:
I think something like the typed test probe `receiveMessages(n, atMost)` is often very useful, because you know what number of events should be produced by some trigger and want to look at all of them.

Also, only providing byte string seems very low level, could we also have something where you pass the parser companion (I think there is a static parser type generated in Java as well) and get a parsed expected proto message, like the Akka `expectMessageType`

Reply:
We do something like the typed parsing here: https://github.com/akka/akka-grpc/pull/1740/files#diff-559fbacacb811f269fb8a1182343f7bc82f934ad952bd0a759a65213d3a2f496R246

Reply:
I must have remembered the Java generated classes part wrong though, no interface there, only static `parseFrom(...)` methods

Reply:
> I think something like the typed test probe `receiveMessages(n, atMost)` is often very useful, because you know what number of events should be produced by some trigger and want to look at all of them.

Yeah, I was trying to keep the API at the minimum while trying this out on samples and seeing what is useful but I agree that one is often very useful. Will add it.

> Also, only providing byte string seems very low level, could we also have something where you pass the parser companion (I think there is a static parser type generated in Java as well) and get a parsed expected proto message, like the Akka expectMessageType

Agree as well. I didn't see a way on how to do it immediately for java without code gen and assumed it was the same for scala.. but good to know it's possible in scala. thanks 👍

Reply:
Maybe a reasonable solution for Java would be a `<T> receiveMessageType(java.util.function.Function<ByteString, T>)` so that users can do something like `receiveMessageType(MyMessage::parseFrom)` with method references? (Didn't try it so maybe the overloads makes that impossible)

Reply:
No, I found it, all generated classes implement `com.google.protobuf.GeneratedMessageV3` which has a `getParserForType` 👍

Reply:
this is added now